### PR TITLE
Add unit tests for divide_and_conquer() - issue 92

### PR DESCRIPTION
**Pull request by AndresSanchezCuadrado**
Unit test divide and conquer static and polymorphic.

### CODE REVIEW

**Review by jdgarciauc3m — CHANGES_REQUESTED** [review]
Remove anything related to polymorphic_execution. That will be handled separately as part of issue #90

File: unit_tests/divideandconquer.cpp [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+// Test for execution policies defined in supported_executions.h
+TYPED_TEST_CASE(divideandconquer_test, executions);
+
+TYPED_TEST(divideandconquer_test, static_empty)
+{
+  this->setup_empty();
+  grppi::divide_and_conquer(this->execution_, this->v, this->out,
+    [this](auto & v) { 
+      // Divide
```
Comment (jdgarciauc3m) [inline unit_tests/divideandconquer.cpp]: Move comment before lambda start

File: unit_tests/divideandconquer.cpp [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+      this->invocations_divide++; 
+      std::vector<std::vector<int> > subproblem;
```
Comment (jdgarciauc3m) [inline unit_tests/divideandconquer.cpp]: Collapse in return std::vector<std::vector<int>>{};

File: unit_tests/divideandconquer.cpp [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+      return subproblem; 
+    },
+    [this](auto & problem, int & out) { 
+      // Base
```
Comment (jdgarciauc3m) [inline unit_tests/divideandconquer.cpp]: Comment before lambda start. Change comment to "solve base case"

File: unit_tests/divideandconquer.cpp [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+      this->invocations_base++; 
+    }, 
+    [this](auto & partial, auto & out) {
```
Comment (jdgarciauc3m) [inline unit_tests/divideandconquer.cpp]: Comment before lambda start. "Combine"

File: unit_tests/divideandconquer.cpp [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+
+
+TYPED_TEST(divideandconquer_test, static_single)
+{
+  this->setup_single();
+  grppi::divide_and_conquer(this->execution_, this->v, this->out,
+    [this](auto & v) { 
+      // Divide
+      this->invocations_divide++; 
+      std::vector<std::vector<int>> subproblem;
```
Comment (jdgarciauc3m) [inline unit_tests/divideandconquer.cpp]: return vector<vector<int>> {v};

File: unit_tests/divideandconquer.cpp [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+
+TYPED_TEST(divideandconquer_test, static_multiple)
+{
+  this->setup_multiple();
+  
+  grppi::divide_and_conquer(this->execution_, this->v, this->out,
+    [this](auto & v) { 
+      // Divide
+      this->invocations_divide++; 
+      std::vector<std::vector<int>> subproblem;
```
Comment (jdgarciauc3m) [inline unit_tests/divideandconquer.cpp]: return vector<vector<int>> {begin(v), end(v)};

**Review by jdgarciauc3m — CHANGES_REQUESTED** [review]
Address comments.

**Comment by AndresSanchezCuadrado** [comment]
Requires to be updated after merged branch of issue #123

**Comment by jdgarciauc3m** [comment]
Issue #123 has already been solved by merging PR #160 to latest

**Comment by jdgarciauc3m** [comment]
Remove tests that are no longer compiling after modifying from directory `legacy_utest`

**Review by jdgarciauc3m — APPROVED** [review]
OK

**Comment by jdgarciauc3m** [comment]
Solves issue #92

**Pull request merged**